drop-in/src/main/java/com/adyen/checkout/dropin/DropIn.kt

```diff
@@ -25,48 +29,201 @@ import com.adyen.checkout.dropin.ui.DropInActivity
  *
  * After setting up the [DropInService], just call [startPayment] and the checkout process will start.
  */
-class DropIn private constructor() {
+object DropIn {
+    private val TAG = LogUtil.getTag()
 
-    companion object {
-        private val TAG = LogUtil.getTag()
+    const val DROP_IN_REQUEST_CODE = 529
 
-        const val RESULT_KEY = "payment_result"
+    const val RESULT_KEY = "payment_result"
+    const val ERROR_REASON_KEY = "error_reason"
+    const val ERROR_REASON_USER_CANCELED = "Canceled by user"
 
-        const val DROP_IN_PREFS = "drop-in-shared-prefs"
-        const val LOCALE_PREF = "drop-in-locale"
+    internal const val DROP_IN_PREFS = "drop-in-shared-prefs"
+    internal const val LOCALE_PREF = "drop-in-locale"
 
-        const val DROP_IN_REQUEST_CODE = 529
+    /**
+     * Starts the checkout flow to be handled by the Drop-in solution.
+     * Make sure you have [DropInService] set up before calling this.
+     * You can pass a [resultHandlerIntent] that will be launched after the Drop-in has completed
+     * without any errors.
+     * We suggest that you set up the [resultHandlerIntent] with the appropriate flags to clear
+     * the stack of the checkout activities.
+     *
+     * 3 states can occur from this operation:
+     * * Cancelled by user: the user dismissed the Drop-in before it has completed.
+     * * Error: a [DropInServiceResult.Error] was returned in the [DropInService], or an error
+     * has occurred.
+     * * Finished: a [DropInServiceResult.Finished] was returned in the [DropInService].
+     *
+     * You should always handle the cases of cancellation and error in [Activity.onActivityResult]
+     * (request code [DROP_IN_REQUEST_CODE]).
+     * You can make use of the [handleActivityResult] helper method to get a [DropInResult] object.
+     * If you prefer to handle the activity result manually, you should expect an
+     * [Activity.RESULT_CANCELED] result code (for both error and cancellation). The data
+     * intent will contain the error reason extra with key [ERROR_REASON_KEY]. Its value will be
+     * [ERROR_REASON_USER_CANCELED] in case of user cancellation or the error reason otherwise.
+     *
+     * As for the Drop-in finished case, if you did not specify a [resultHandlerIntent], you will
+     * also receive the result in [Activity.onActivityResult]. You can make use of the
+     * [handleActivityResult] helper method. If you prefer to handle the activity result
+     * manually, you should expect an [Activity.RESULT_OK] result code. The data intent will
+     * contain the result string extra with key [RESULT_KEY].
```

Comment:
Maybe also explain that the content of `RESULT_KEY` will be what they specified and sent back in the `DropInService`?